packages/core-v2-graphql/src/models/Token.ts

```diff
@@ -1,4 +1,67 @@
+import { ERC20__factory } from "@elementfi/core-v2-typechain";
 import { CoreV2Context } from "src/context";
+import { TokenContract } from "src/datasources/TokenContract";
 import { Token } from "src/generated";
+import { getDataSourceByAddress } from "src/utils/getDataSourceByAddress";
 
-export const TokenModel = {};
+// model should be a class that takes in a datasource(s)
+export const TokenModel = {
+  getAllowance,
+  getBalanceOf,
+  getByAddress,
+};
+
+interface GetByAddressOptions {
+  address: string;
+  context: CoreV2Context;
+}
+
+async function getByAddress({
+  address,
+  context,
+}: GetByAddressOptions): Promise<Token> {
+  const tokenContract = getDataSourceByAddress(
+    address,
+    context.elementDataSources,
+    "tokenContracts",
+  );
+
+  return {
+    address: await tokenContract.address,
+    decimals: await tokenContract.decimals(),
+    symbol: await tokenContract.symbol(),
+    name: await tokenContract.name(),
```

Comment:
or maybe `tokenContract.getInfo()`

Reply:
It kinda feels weird to me to put methods on something called `tokenContract` that aren't on the actual contract.  Is there a better way to do that?

Reply:
it will be one call to the datasource object, but still N calls to the smart contract. This is an opportunity to use a multicall contract to batch these calls together. Was going to talk to you about this since it involves some complexity so we get down a standard.

Reply:
@sentilesdal I'm open to leaving them single calls, but they're abstractions of the contract. They don't need to match the contract methods 1:1. They're could be variations as well like there are for the VotingVaultContract data source. It's one API layer over multiple varieties. While the contract methods are designed for efficiency, the data source abstractions are designed for use by other scripts. We could change the name to `TokenDataSource` to separate them further.

@cashd That sounds super interesting.